src/playground/sample_data_ingestion/utils/sample_data_ingestor/sample_data_ingestor.py

```diff
@@ -0,0 +1,236 @@
+        entry = generate_data(input, mapping) 
+        milliseconds = int(current_date.strftime("%s")) * 1000
+        # If entry came from an input file, load data to the dataset list
+        if type(entry) is list:
+            for element in entry:
+                element = loads(element)
+                element[feature] = generate_noise(element[feature], anomaly_percentage, avg_min, avg_max, abs_min, abs_max)
+                element[timestamp] = milliseconds
+                dataset.append(element)
+        else:
+            entry = loads(entry)
+            entry[feature] = generate_noise(entry[feature], anomaly_percentage, avg_min, avg_max, abs_min, abs_max)
+        
+            # Update timestamp with current time
+            entry[timestamp] = milliseconds
+            dataset.append(dumps(entry))
+        # Increment current date
+        current_date += timedelta(minutes = minutes)
+    return dataset
+
+
+"""
+Function that ingests user-provided data or generated data
+Arguments:  
+    - template: JSON file, CSV file, JSON mapping, or JSON short-hand template
+    - index_name: The name of the index to ingest data
+    - mapping: Whether or not the input is a mapping
+    - file_provided: Boolean flag as to whether or not a file was provided
+    - chunk: How many documents can be ingested per BULK call
+    - feature: The field that will be used to test anomalies
+    - timestamp: The field name that contains timestamps
+    - number: How many documents to generate
+    - avg_min: The lowest average value in which there is no anomaly
+    - avg_max: The highest average value in which there is no anomaly
+    - abs_min: The lowest possible value, which can be an anomaly
+    - abs_max: The highest possible value, which can be an anomaly
+    - anomaly_percentage: How often an anomaly occurs (minimum: 0.001)
+    - minutes: The time interval for each data point (e.g. if minutes = 2, this tool will generate entries with timestamps that are 2 minutes apart from one another)
+    - current_date: The date at which entries are generated
+"""
+def ingest(client, 
+    template,
+    index_name,
+    file_provided = False,  
+    mapping = True,  
+    number = 6, 
+    chunk = 5, 
+    timestamp = None,
+    feature = None,  
+    avg_min = 0, 
+    avg_max = 0, 
+    abs_min = 0, 
+    abs_max = 0, 
+    anomaly_percentage = 0, 
+    minutes = 2,
+    current_date = datetime.today()
+):
+    # Various errors for invalid arguments
+    if not template or not index_name:
+        raise ValueError("You must have a template and an index name to continue")
+    if not client:
+        raise ValueError("Client needed to connect")
+    if avg_min < abs_min or avg_max > abs_max or avg_min > avg_max or abs_min > abs_max:
+        raise ValueError("Check ranges for generating trend data")
+    if anomaly_percentage < 0 or anomaly_percentage > 1:
+        raise ValueError("Anomaly percentage cannot be outside a percent")
```

Comment:
Let's decouple input validation from ingestion logic

Reply:
Done, the error raising has been refactored to `ingest_validation()`